### PR TITLE
feat: add commit list view in source control

### PR DESCRIPTION
**Pull request by conwnet**
1. Support view/switch commits in source control panel.
2. Add **Refresh** button to `pull list view` and `commit list view`

### CODE REVIEW

**Comment by conwnet** [comment]
related with #236

File: extensions/github1s/src/views/commit-list-view.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+	public updateTree() {
+		this.forceUpdate = true;
+		this._onDidChangeTreeData.fire(undefined);
+	}
+
+	async getCommitItems(): Promise<CommitTreeItem[]> {
+		const { ref } = await router.getState();
+		// only recent 100 commits will be list here
+		// TODO: implement pagination
```
Comment (xcv58) [inline extensions/github1s/src/views/commit-list-view.ts:45]: Do you have any idea of how we could support pagination in the future?

I think one possible way is to add an option at the bottom to `load more commits` and it triggers another call for the next page and update the quick pick list.

**Review by xcv58 — APPROVED** [review]
The code & functionality looks good to me. Just add some nits and thought.

**Comment by conwnet** [comment]
@conwnet  Thanks you so much for the review! I have create a [issue](https://github.com/conwnet/github1s/issues/255) to note these problems. I will fixed them later.❤️

File: extensions/github1s/src/commands/commit.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+export const commandSwitchToCommit = async (commitSha?: string) => {
+	const { owner, repo } = await router.getState();
+
+	// if the a commitSha isn't provided, use quickInput
+	if (!commitSha) {
+		// manual input a commit sha
+		const inputCommitShaItem: vscode.QuickPickItem = {
+			label: '$(git-commit) Manual input the commit sha',
+			alwaysShow: true,
+		};
```
[the remark this answers is not in the document]
Comment (conwnet) [reply extensions/github1s/src/commands/commit.ts:20]: Yes, we have a similar command to do this: https://github.com/conwnet/github1s/blob/bbc88fee5307a48128be0753f481c1b21a549b9b/extensions/github1s/package.json#L107-L112

File: extensions/github1s/src/commands/commit.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+		// use the commit list as the candidates
+		const commitItems: vscode.QuickPickItem[] = (
+			await repository.getCommits((await router.getState()).ref)
+		).map((commit) => ({
+			commitSha: commit.sha,
+			label: `(${commit.sha.slice(0, 7)}) ${commit.commit.message}`,
```
[the remark this answers is not in the document]
Comment (conwnet) [reply extensions/github1s/src/commands/commit.ts:26]: Good idea! I will put more infomation to the PR/Commit select items~

File: extensions/github1s/src/commands/commit.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+			commitSha = await vscode.window.showInputBox({
+				placeHolder: 'Please input the commit sha',
+			});
+		} else {
+			// select a commit sha
+			commitSha = choice.commitSha;
+		}
+	}
+
+	commitSha && router.replace(`/${owner}/${repo}/commit/${commitSha}`);
```
[the remark this answers is not in the document]
Comment (conwnet) [reply extensions/github1s/src/commands/commit.ts:55]: Yes, I think we should do this, and the same for PR.

File: extensions/github1s/src/listeners/router/explorer.ts [hunk 5]
```diff
@@ -19,7 +19,8 @@ export const explorerRouterListener = (
 	if (
 		currentState.owner !== previousState.owner ||
 		currentState.repo !== previousState.repo ||
-		currentState.ref !== previousState.ref
+		currentState.ref !== previousState.ref ||
+		currentState.pageType !== previousState.pageType
```
[the remark this answers is not in the document]
Comment (conwnet) [reply extensions/github1s/src/listeners/router/explorer.ts:23]: You are right! Sorry I forget this way when I write here.

File: extensions/github1s/src/views/commit-list-view.ts (hunk 1 above)
Comment (conwnet) [reply extensions/github1s/src/views/commit-list-view.ts:45]: Yes, I agree it, I also want add a special item named `Load more` below these items~

**Review by Siddhant-K-code — COMMENTED** [review]
Good Work @conwnet <3

**Pull request merged**